src/main/java/com/spikes2212/util/smartmotorcontrollers/TalonFXWrapper.java

```diff
@@ -0,0 +1,138 @@
+package com.spikes2212.util.smartmotorcontrollers;
+
+import com.ctre.phoenix6.CANBus;
+import com.ctre.phoenix6.configs.*;
+import com.ctre.phoenix6.controls.*;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.hardware.core.CoreTalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import com.spikes2212.control.FeedForwardSettings;
+import com.spikes2212.control.PIDSettings;
+import com.spikes2212.control.TrapezoidProfileSettings;
+import com.spikes2212.util.UnifiedControlMode;
+
+public class TalonFXWrapper implements SmartMotorController {
+
+    protected final TalonFX talonFX;
+    protected final Slot0Configs closedLoopConfig;
+    protected final MotorOutputConfigs motorOutputConfigs;
+
+    public TalonFXWrapper(int deviceId, String canbus) {
+        talonFX = new TalonFX(deviceId, canbus);
+        closedLoopConfig = new Slot0Configs();
+        motorOutputConfigs = new MotorOutputConfigs();
+    }
+
+    public TalonFXWrapper(int deviceId, CANBus canbus) {
+        this(deviceId, canbus.getName());
+    }
+
+    public TalonFXWrapper(int deviceId) {
+        this(deviceId, "");
+    }
+
+    public void restoreFactoryDefaults() {
+        talonFX.getConfigurator().apply(new TalonFXConfiguration());
+        motorOutputConfigs.withInverted(InvertedValue.CounterClockwise_Positive);
+    }
+
+    public void setIdleMode(NeutralModeValue neutralModeValue) {
+        talonFX.getConfigurator().apply(motorOutputConfigs.withNeutralMode(neutralModeValue));
+    }
+
+    public TalonFXConfigurator getConfigurator() {
+        return talonFX.getConfigurator();
+    }
+
+    @Override
+    public void configurePID(PIDSettings pidSettings) {
+        closedLoopConfig.kP = pidSettings.getkP();
+        closedLoopConfig.kI = pidSettings.getkI();
+        closedLoopConfig.kD = pidSettings.getkD();
+        talonFX.getConfigurator().apply(closedLoopConfig);
+    }
+
+    @Override
+    public void configureFF(FeedForwardSettings feedForwardSettings) {
```

Comment:
configure the GravityType and StaticFeedForwardSignValue according to the settings' ControlMode